### DISCUSSION TITLE
pimcore_inc with snippets has a changed behavior

### DISCUSSION THREAD

**Discussion by akno77** [pimcore/pimcore#16141] [Q&A / Support]
Hi,

in Pimcore 10 it was possible to have a pimcore_inc call like this in the layout template:

`{{ pimcore_inc(document.getProperty('footer') , {'snippetParams': {'testParam': 'test Value'}}) }}`

The controller action looked like this:

```class DefaultController extends FrontendController
{
    /**
     * @Template
     * @param Request $request
     * @return array
     */
    public function defaultAction(Request $request)
    {
        // handle params passed to snippets to be accessible in template
        $snippetParams = $request->get('snippetParams');

        return [
            'snippetParams' => $snippetParams
        ];
    }
}
```

The snippet document could have now the "DefaultController" assigned and any template you want.

In the snippet template you could access then the global Twig var "snippetParams".

In Pimcore 11 this doesn't work anymore. If you assign directly a template in the Snippet Document settings the returned template vars in the action are not present. If you define its own controller action for the snippet with the template defined in the action and you leave the template option in the document settings empty, then it works.

Is this a wanted changed behavior in Pimcore 11 or is it a bug?

If it is not a bug the whole concept with using any controller action but different templates doesn't work anymore...

**Comment by NiklasBr** [comment]
Also experiencing this with a site relying heavily on it and could not find any documentation on it.

**Reply by akno77** [reply]
Has anyone from the Pimcore team an answer if this is a bug or feature?
